flux_led/__main__.py

```diff
@@ -741,22 +824,56 @@ def setRgbw(self, r,g,b,w, persist=True, brightness=None, retry=2):
         else:
             msg = bytearray([0x41])
 
-        msg.append(int(r))
-        msg.append(int(g))
-        msg.append(int(b))
-        msg.append(int(w))
-        if self.protocol == "LEDENET":
+        if r is not None:
+            msg.append(int(r))
+        else:
+            msg.append(int(0))
+        if g is not None:
+            msg.append(int(g))
+        else:
+            msg.append(int(0))
+        if b is not None:
+            msg.append(int(b))
+        else:
+            msg.append(int(0))
+        if w is not None:
             msg.append(int(w))
+        else:
+            msg.append(int(0))
+
+        # LEDENET devices support two white outputs for cool and warm. We set
+        # the second one here - if we're only setting a single white value,
+        # we set the second output to be the same as the first
+        if self.protocol == "LEDENET":
+            if w2 is not None:
+                msg.append(int(w2))
+            elif w is not None:
+                msg.append(int(w))
+            else:
+                msg.append(0)
 
         if not self.rgbwprotocol:
-            if w > 0:
+            # For devices that can't set RGB+W simultaneously, indicate whether
+            # we should set the white outputs or the RGB outputs
+            if w is not None or w2 is not None:
                 msg.append(0x0f)
             else:
                 msg.append(0xf0)
         else:
+            # These devices can simultaneously control the RGB and white output
             msg.append(0x00)
 
+        # Message terminator
         msg.append(0x0f)
+
+        # yet another type of protocol/controller that only supports RGB (no ww)
+        if self.protocol == 'LEDnet_wifi370':
+            msg = bytearray([0x56])
+            msg.append(int(r))
+            msg.append(int(g))
+            msg.append(int(b))
+            msg.append(0xaa)
+        
```

Comment:
We've gone to all the work of creating a msg buffer already at this point, and then we throw away and start it again if it's one of these devices. It'd be better to add this at the top and then put the existing code as the other branch of an else:, otherwise it's confusing as to what's going on (also this line has additional spaces in it)